python/thunder/imgprocessing/transformation.py

```diff
@@ -8,6 +9,168 @@ def apply(self, im):
         raise NotImplementedError
 
 
+class DifferentiableTransformation(Transformation):
```

Comment:
Maybe rather than subclass `Transformations` make this more like a mix-in, called `Differentiable`, then the other two can be `AffineTransformation(Transformation, Differentiable)` and `EuclideanTransformation(AffineTransformation)`? I guess it depends a little on what else you have in mind for this. But in terms of what it contributes `Differentiable` seems more like a mix-in than a subclass to me.